plugin/documents.py

```diff
@@ -328,6 +328,8 @@ def session_views_async(self) -> list[SessionView]:
         return list(self._session_views.values())
 
     def on_text_changed_async(self, change_count: int, changes: Iterable[sublime.TextChange]) -> None:
+        if not self.sessions_async():
+            return
         if self.view.is_primary():
             for sv in self.session_views_async():
                 sv.on_text_changed_async(change_count, changes)
```

Comment:
So I guess this could be considered a bit more efficient since we don't have to get sessions twice.
The con is that the code has to be case-specific and be less consistent.

```suggestion
    def on_text_changed_async(self, change_count: int, changes: Iterable[sublime.TextChange]) -> None:
        sessons = self.session_views_async()
        if not sessions:
            return
        if self.view.is_primary():
            for sv in sessions:
                sv.on_text_changed_async(change_count, changes)
```

The alternative that would look clean is to have a decorator that would be attached to a method. But that's obviously less flexible and wouldn't allow us to create fully optimized early exit.

Reply:
Actually we can always check for SessionViews, because `self.sessions_async` uses `self.session_buffers_async` -> `self.session_views_async` anyway. https://github.com/sublimelsp/LSP/pull/2532/commits/7a8b4821c38f5a4de25e021963035d67429e684f

I do like the idea with the decorator though.